### PR TITLE
Fit 2 phases

### PR DESCRIPTION
**Pull request by DarioZappala**
# What's new
- Class `GaussNewton_MultiPhases` to perform Gauss-Newton minimisation on multiple phases that share calibration parameters `s`, `beta` and have separate `a`.
- `GaussNewton.theta_range()` is now a property, to be in line with `GaussNewton_MultiPhases.theta_range()`.
- Class `GammaMap_MultiPhases`, that represents a map with an instance of `GaussNewton_MultiPhases` for each pixel. This class is derived from `GammaMap`.

### CODE REVIEW

File: XRDXRFutils/gaussnewton_multi.py [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+
+    def der_f_tau(self):
+        mu = [arr[newaxis, :] for arr in self.mu]
+        sigma2 = [arr[newaxis, :] for arr in self.sigma2]
+        theta = [arr[:, newaxis] for arr in self.theta]
+        return [self.component_full[idx] * ((theta[idx] - mu[idx])**2 / (2 * sigma2[idx]**2)) * self.der_u(self.tau[idx]) for idx in range(self.n_phases)]
+
+
+    def fit(self, a = False, s = False, beta = False, gamma = False, sigma = False, alpha = 1, downsample = None):
```
Comment (zpreisler) [inline XRDXRFutils/gaussnewton_multi.py:241]: You should have a separated into a1 and a2 here. They are equivalent to s, beta and gamma. Currently, I can not fix one phase and iterate only the other one If I need to or at least I do not see how.

Comment (DarioZappala) [reply XRDXRFutils/gaussnewton_multi.py:241]: Yes, I see your point. I'm going to deal with it.

Comment (zpreisler) [reply XRDXRFutils/gaussnewton_multi.py:241]: ok thanks

Comment (DarioZappala) [reply XRDXRFutils/gaussnewton_multi.py:251]: You can test the new joined calibration in [this notebook](https://github.com/DarioZappala/XRDAnalysis/blob/master/XRD_1p_test_phase_shift__joined_fit.ipynb).

Comment (zpreisler) [reply XRDXRFutils/gaussnewton_multi.py:251]: In the notebook:
Calculate the shift in microns along the direction of the incident ray. Apply the formula from https://codimd.infn.it/TGkIwbqqTEqHft9AmigaEw# and then calculate the new z in microns.

Comment (DarioZappala) [reply XRDXRFutils/gaussnewton_multi.py:264]: Done. Now I calculate Delta y (along the incident direction) and Delta z (along the direction orthogonal to the sample surface).
I've also updated in the same way the two notebooks that search for stratigraphy in a map with separate calibration ([here](https://github.com/DarioZappala/XRDAnalysis/blob/master/XRD_2d_phase_shift__separate_fit.ipynb)) and with joined calibration ([here](https://github.com/DarioZappala/XRDAnalysis/blob/master/XRD_2d_phase_shift__joined_fit.ipynb)).

**Pull request merged**